packages/dart_frog/lib/src/serve.dart

```diff
@@ -26,5 +27,6 @@ Future<HttpServer> serve(
     port,
     poweredByHeader: poweredByHeader,
     securityContext: securityContext,
+    shared: shared,
```

Comment:
Can we add some tests to verify it defaults to false and that is being passed as expected? 

Inspiration can be taken from the test group [X-Powered-By-Header](https://github.com/VeryGoodOpenSource/dart_frog/blob/main/packages/dart_frog/test/src/serve_test.dart#L47) at the [serve_test.dart](https://github.com/VeryGoodOpenSource/dart_frog/blob/main/packages/dart_frog/test/src/serve_test.dart#L47) file.

Reply:
Would that be as simple as me checking the shared property on the server or would I check if multithreading it works properly?

Reply:
I think the former would be enough since the latter should be tackled by shelf.

Reply:
I have to say I feel a little silly but I can't seem to find a way to do the prior.
`      test('is configured by default', () async {
        try {
          final server = await serve((_) => Response(), 'localhost', 3000);
          final testServer = await serve((_) => Response(), 'localhost', 3000);
          await server.close();
          await testServer.close();
        } catch (e) {
          expect(
            e.toString(),
            equals(
              SocketException(
                'Failed to create server socket',
                osError: OSError(
                  'The shared flag to bind() needs to be `true` if binding multiple times on the same (address, port) combination.',
                ),
                address: (e as SocketException).address,
                port: 3000,
              ).toString(),
            ),
          );
        }
      });`
      Toying with this but even then I cannot close close the servers obbviously.

Reply:
Thanks @femalemonkeyman for adding tests! Do not feel silly about this 💙 . I've added a few suggestions to the tests, let me know what you think 👀 🙌